### PR TITLE
pkg/cmsis-dsp: Add CMSIS-DSP as a package

### PR DESCRIPTION
**Pull request by gebart**
CMSIS-DSP is the ARM provided math library optimized for Cortex-M processors. It can be downloaded from http://www.arm.com/products/processors/cortex-m/cortex-microcontroller-software-interface-standard.php after a free registration. The library sources are licensed under a BSD 3-clause license which makes it possible to redistribute them outside the registration wall. I put a clone of the sources at https://github.com/gebart/CMSIS-DSP for easier (scripted) access.

The library could probably be modified in a way that it compiles on non-Cortex-M platforms (defining ARM_MATH_CM0 for a non-Cortex platform would make the larger part of the library compile without errors on any platform), but this PR is limited to the Cortex-M based boards.

### CODE REVIEW

**Comment by gebart** [comment]
Added a small test case which is white listed only for the Cortex boards.

**Comment by gebart** [comment]
Rebased

**Comment by gebart** [comment]
needs rebase and refactoring once #3023 is merged.

**Comment by gebart** [comment]
rebased and refactored to fit with the cortexm_common framework

**Comment by gebart** [comment]
Needs an updated CMSIS  v 4.3 in order to build for CM0 #3128

**Comment by gebart** [comment]
Needs #3099 for mulle and pba-d-01-kw2x

**Comment by gebart** [comment]
rebased since #3128 was merged.

**Comment by gebart** [comment]
rebased after #3099 was merged

**Comment by gebart** [comment]
Ready for review

**Comment by gebart** [comment]
ping @kaspar030

File: tests/pkg_cmsis-dsp/Makefile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  msbiot \
+  mulle \
+  nucleo-f091 \
+  nucleo-f334 \
+  nucleo-l1 \
+  openmote \
+  pba-d-01-kw2x \
+  pca10000 \
+  pca10005 \
+  samr21-xpro \
```
Comment (cgundogan) [inline tests/pkg_cmsis-dsp/Makefile]: Is there any reason why saml21-xpro is not listed here?

Comment (gebart) [inline tests/pkg_cmsis-dsp/Makefile]: this PR is too old, rebasing.

**Comment by gebart** [comment]
Addressed comment by @cgundogan , added two new cortex boards to the list.
(use `for f in *; do git grep -l cortex $f | cat; done | cut -d/ -f1 | sort | uniq | awk '{ print "  "$0" \\"; }'` to get a list)

**Comment by gebart** [comment]
OK to squash?

**Comment by cgundogan** [comment]
Although I have no hardware avail. right now to test this, I can confirm that compiling the included test works at least. I'd say just go ahead and squash, but I would wait for someone with more experience than me before merging. @OlegHahm @kaspar030 ?

**Comment by gebart** [comment]
squashed.

**Comment by PeterKietzmann** [comment]
Nice! I'm really happy about this support. The output of the test application (running on stm32f4discovery) seems correct.
```
2015-06-25 08:47:01,095 - INFO # kernel_init(): This is RIOT! (Version: 2014.05-3902-g2f47-tes_gebart_dsp)
2015-06-25 08:47:01,099 - INFO # kernel_init(): jumping into first task...
2015-06-25 08:47:01,100 - INFO # ARM DSP lib test
2015-06-25 08:47:01,109 - INFO # Note: This test is using 32 bit IEEE 754 floating point numbers,(24 bit mantissa, 7 bit exponent)
2015-06-25 08:47:01,113 - INFO # Expect roughly 7 decimals precision on the result.
2015-06-25 08:47:01,115 - INFO #       dsp:   0.903941760
2015-06-25 08:47:01,117 - INFO # reference:   0.903941793931839
2015-06-25 08:47:01,120 - INFO #      diff:   0.000000060
2015-06-25 08:47:01,121 - INFO # Test done
```

**Comment by gebart** [comment]
@PeterKietzmann That is the same output that I have as well.

**Comment by PeterKietzmann** [comment]
That is what I have assumed :-) . It was more meant to show @cgundogan how the output looks like

**Comment by PeterKietzmann** [comment]
Is the build-test failing for cortex_m4 expected? (compare travis output)

**Comment by gebart** [comment]
@PeterKietzmann 
```
W: Failed to fetch http://us.archive.ubuntu.com/ubuntu/dists/trusty-updates/main/binary-amd64/Packages  Hash Sum mismatch

W: Failed to fetch http://us.archive.ubuntu.com/ubuntu/dists/trusty-updates/main/binary-i386/Packages  Hash Sum mismatch

E: Some index files failed to download. They have been ignored, or old ones used instead.
```
Travis messed up

**Comment by PeterKietzmann** [comment]
I think I'm not in the position to final ACK and merge this PR. Anyway, it is reviewed by @cgundogan (is it) and roughly tested by me and travis. Who is willing to push the button?

**Comment by gebart** [comment]
Any ACKs?
ping @kaspar030 @OlegHahm

**Comment by PeterKietzmann** [comment]
Code looks good, example works as expected, commits look good. Even if I'm not too deep inside this PR I say ACK and go

**Pull request merged**